hibernate-reactive-core/src/test/java/org/hibernate/reactive/BeforeExecutionIdGeneratorTypeTest.java

```diff
@@ -0,0 +1,107 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive;
+
+import org.hibernate.annotations.IdGeneratorType;
+import org.hibernate.generator.EventType;
+import org.hibernate.reactive.id.ReactiveIdentifierGenerator;
+import org.hibernate.reactive.session.ReactiveConnectionSupplier;
+import org.hibernate.reactive.util.impl.CompletionStages;
+
+import org.junit.jupiter.api.Test;
+
+import io.vertx.junit5.Timeout;
+import io.vertx.junit5.VertxTestContext;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@Timeout(value = 10, timeUnit = MINUTES)
+public class BeforeExecutionIdGeneratorTypeTest extends BaseReactiveTest {
+
+	@Override
+	protected Collection<Class<?>> annotatedEntities() {
+		return List.of( Person.class );
+	}
+
+	@Test
+	public void testPersist(VertxTestContext context) {
+		Person person = new Person();
+		test(
+				context, openSession()
```

Comment:
Could you use `getMutinySessionFactory.withTransaction`?
I think it's more similar to a typical use case.

Maybe we could also test without transactions, but we should at least add a comment to explain what's going on

Reply:
Would it make sense to also read the value from the database to make sure  that they match? Maybe it's overkill